nextjs-13/src/components/ListNewsletter/index.tsx

```diff
@@ -0,0 +1,34 @@
+interface Record {
+  id: string;
+  name: string;
+}
+
+interface PromiseWrapper {
+  read: () => Record[];
+}
+
+interface Props {
+  promise: PromiseWrapper;
+}
+
+const ListNewsletter = ({ promise }: Props) => {
```

Comment:
Why pass a promise? That is kind of unexpected as a prop. The most standard way would be to pass an array of newsletters. It adds complexity for testing the component (https://github.com/nimblehq/js-framework-benchmark/pull/110/files#diff-fbbd0021aabad22d799d758d4fc16f2c26225f683f50b56c7cbbb068cbb1c774R22) for no apparent benefits